### PR TITLE
Onboarding does not work without chart title

### LINKED ISSUE #114: onboarding does not work without a title
- [x] plotly treemap

### PR DESCRIPTION
**Pull request by dvsilviyageo**
closes #114 
Summary:
* In the core package of each chart, where the messages are added, it is checked whether the chart has title if not 'chartTitle' is not added to the message.

screenshots:
image

### CODE REVIEW

File: packages/core/src/bar-chart.ts [hunk 1]
```diff
@@ -16,76 +24,143 @@ export interface IOnboardingBarChartSpec extends IOnboardingSpec {
...
+  const reading = defaultOnboardingStages.get(
+    EDefaultOnboardingStages.READING
+  ) as IOnboardingStage;
+  const interacting = defaultOnboardingStages.get(
+    EDefaultOnboardingStages.USING
+  ) as IOnboardingStage;
+  let messages: IOnboardingMessage[];
+
+  if (spec.chartTitle?.value === undefined) {
+    messages = [
```
Comment (dvdanielamoitzi) [inline packages/core/src/bar-chart.ts:41]: this is lots of code duplication. please create the messages array outside the if and then just push the title message inside the if to the messages array

File: packages/core/src/change-matrix.ts [hunk 2]
```diff
@@ -9,61 +17,105 @@ export interface IOnboardingChangeMatrixSpec extends IOnboardingSpec {
...
-      text: `The columns show the ${spec.xAxis?.value}, while the rows show the ${spec.yAxis?.value}.`,
-      title: 'Reading the chart',
-      onboardingStage: reading,
-      marker: {
-        id: "unique-marker-id-4"
-      }
-    },
-  ];
+  if (spec.chartTitle?.value === undefined) {
+    messages = [
```
Comment (dvdanielamoitzi) [inline packages/core/src/change-matrix.ts:30]: same here. please create the array with common messages outside the if and only push the other messages to this array inside the if

File: packages/core/src/horizon-graph.ts [hunk 3]
```diff
@@ -10,93 +17,180 @@ export interface IOnboardingHorizonGraphSpec extends IOnboardingSpec {
...
+  const reading = defaultOnboardingStages.get(
+    EDefaultOnboardingStages.READING
+  ) as IOnboardingStage;
+  const using = defaultOnboardingStages.get(
+    EDefaultOnboardingStages.USING
+  ) as IOnboardingStage;
+  let messages: IOnboardingMessage[];
+
+  if (spec.chartTitle?.value === undefined) {
+    messages = [
```
Comment (dvdanielamoitzi) [inline packages/core/src/horizon-graph.ts:37]: same here

File: packages/core/src/scatterplot.ts [hunk 4]
```diff
@@ -12,78 +19,145 @@ export interface IOnboardingScatterplotSpec extends IOnboardingSpec {
...
-      onboardingStage: reading,
-      marker: {
-        id: "unique-marker-id-6"
-      }
-    }
-  ];
+  if (spec.chartTitle?.value === undefined) {
+    messages = [
+      {
+        anchor: getAnchor(spec.type, visElement),
```
Comment (dvdanielamoitzi) [inline packages/core/src/scatterplot.ts:40]: same here

**Review by dvdanielamoitzi — CHANGES_REQUESTED** [review]

File: packages/core/src/treemap.ts [hunk 5]
```diff
@@ -10,106 +17,196 @@ export interface IOnboardingTreemapSpec extends IOnboardingSpec {
...
-      requires: ['gapDesc'],
-      text: `Items within a sub-category are represented by rectangles that are closely packed together with increasingly larger gaps to the neighboring categories.`,
-      title: 'Reading the chart',
-      onboardingStage: reading,
-      marker: {
-        id: "unique-marker-id-5"
-      }
-    },
+  if (spec.chartTitle?.value === undefined) {
+    messages = [
```
[the remark this answers is not in the document]
Comment (dvsilviyageo) [reply packages/core/src/treemap.ts]: I have refactored the code to remove the code duplication and made a new commit

File: packages/core/src/horizon-graph.ts [hunk 6]
```diff
@@ -10,93 +17,110 @@ export interface IOnboardingHorizonGraphSpec extends IOnboardingSpec {
...
+  visElement: Element
+): IOnboardingMessage[] {
+  const reading = defaultOnboardingStages.get(
+    EDefaultOnboardingStages.READING
+  ) as IOnboardingStage;
+  const using = defaultOnboardingStages.get(
+    EDefaultOnboardingStages.USING
+  ) as IOnboardingStage;
+  let messages: IOnboardingMessage[];
+  messages = [
```
Comment (dvdanielamoitzi) [inline packages/core/src/horizon-graph.ts:35]:
```suggestion
 const messages: IOnboardingMessage[] = [
```

File: packages/core/src/scatterplot.ts [hunk 7]
```diff
@@ -12,78 +19,93 @@ export interface IOnboardingScatterplotSpec extends IOnboardingSpec {
...
+  const analyzing = defaultOnboardingStages.get(
+    EDefaultOnboardingStages.ANALYZING
+  ) as IOnboardingStage;
+  const reading = defaultOnboardingStages.get(
+    EDefaultOnboardingStages.READING
+  ) as IOnboardingStage;
+  const interacting = defaultOnboardingStages.get(
+    EDefaultOnboardingStages.USING
+  ) as IOnboardingStage;
+  let messages: IOnboardingMessage[];
```
Comment (dvdanielamoitzi) [inline packages/core/src/scatterplot.ts:35]:
```suggestion
  ) as IOnboardingStage;
```

File: packages/core/src/scatterplot.ts [hunk 8]
```diff
@@ -12,78 +19,93 @@ export interface IOnboardingScatterplotSpec extends IOnboardingSpec {
...
-      anchor: getAnchor(spec.chartTitle, visElement),
-      requires: ['chartTitle'],
-      text: `The chart shows the ${spec.chartTitle?.value}.`,
-      title: 'Reading the chart',
-      onboardingStage: reading,
-      marker: {
-        id: "unique-marker-id-1"
-      }
-    },
+  messages = [
```
Comment (dvdanielamoitzi) [inline packages/core/src/scatterplot.ts:37]:
```suggestion
  const messages: IOnboardingMessage[] = [
```

File: packages/core/src/treemap.ts [hunk 9]
```diff
@@ -10,106 +17,120 @@ export interface IOnboardingTreemapSpec extends IOnboardingSpec {
...
+    EDefaultOnboardingStages.ANALYZING
+  ) as IOnboardingStage;
+  const reading = defaultOnboardingStages.get(
+    EDefaultOnboardingStages.READING
+  ) as IOnboardingStage;
+  const interacting = defaultOnboardingStages.get(
+    EDefaultOnboardingStages.USING
+  ) as IOnboardingStage;
+  let messages: IOnboardingMessage[];
+  messages = [
```
Comment (dvdanielamoitzi) [inline packages/core/src/treemap.ts:38]:
```suggestion
  const messages: IOnboardingMessage[] = [
```

**Review by dvdanielamoitzi — CHANGES_REQUESTED** [review]

**Review by dvdanielamoitzi — APPROVED** [review]

**Pull request merged**